File: spring-web-java/src/main/java/spring/web/java/global/common/DateTimeUtils.java
package spring.web.java.global.common;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;
import java.time.ZoneId;
import java.time.ZoneOffset;

public abstract class DateTimeUtils {

	public static final ZoneId KST_ZONE = ZoneId.of("Asia/Seoul");

	public static LocalDateTime convertUtcToKst(LocalDateTime utcDateTime) {
		return utcDateTime.atZone(ZoneOffset.UTC).withZoneSameInstant(KST_ZONE).toLocalDateTime();
	}

	public static LocalDateTime convertKstToUtc(LocalDateTime kstDateTime) {
		return kstDateTime.atZone(KST_ZONE).withZoneSameInstant(ZoneOffset.UTC).toLocalDateTime();
	}

	public static OffsetDateTime toOffsetDateTime(LocalDateTime localDateTime) {
		return localDateTime.atZone(KST_ZONE).toOffsetDateTime();
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/global/common/DateTimeUtils.kt
package spring.web.kotlin.global.common

import java.time.LocalDateTime
import java.time.OffsetDateTime
import java.time.ZoneId
import java.time.ZoneOffset

object DateTimeConstants {
    val KST_ZONE: ZoneId = ZoneId.of("Asia/Seoul")
}

fun LocalDateTime.convertToKst(): LocalDateTime =
    this.atZone(ZoneOffset.UTC).withZoneSameInstant(DateTimeConstants.KST_ZONE).toLocalDateTime()

fun LocalDateTime.convertToUtc(): LocalDateTime =
    this.atZone(DateTimeConstants.KST_ZONE).withZoneSameInstant(ZoneOffset.UTC).toLocalDateTime()

fun LocalDateTime.toOffsetDateTime(): OffsetDateTime =
    this.atZone(DateTimeConstants.KST_ZONE).toOffsetDateTime()
